src/main/java/de/azapps/kafkabackup/sink/BackupSinkTask.java

```diff
@@ -11,23 +12,30 @@
 import org.apache.kafka.common.TopicPartition;
 import org.apache.kafka.connect.sink.SinkRecord;
 import org.apache.kafka.connect.sink.SinkTask;
+import org.apache.kafka.connect.sink.SinkTaskContext;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import java.io.IOException;
 import java.nio.file.Files;
 import java.nio.file.Path;
 import java.nio.file.Paths;
-import java.util.Collection;
-import java.util.HashMap;
-import java.util.Map;
+import java.util.*;
 
 public class BackupSinkTask extends SinkTask {
     private static final Logger log = LoggerFactory.getLogger(BackupSinkTask.class);
     private Path targetDir;
     private Map<TopicPartition, PartitionWriter> partitionWriters = new HashMap<>();
     private long maxSegmentSizeBytes;
     private OffsetSink offsetSink;
+    private SinkTaskContext sinkTaskContext;
```

Comment:
why do we need this (and the other variables apart from the adminClient) as instance vars?

Reply:
I removed some instance variables, but retained most of them. I need to keep track of offsets and be able to override stuff for testing.